File: insertion-sort/insertion-sort.py
# Insertion sort in Python

def insertionSort(array):
    
    for step in range(1, len(array)):
        key = array[step]
        j = step - 1

        while j > 0 and array[j] > key:
            array[j + 1] = array[j]
            j = j - 1
        array[j + 1] = key
    
    print(array)

insertionSort([2,3,64,754,56,6])

File: insertion-sort/insertion-sort.rs
// Insertion sort 

pub fn insertion_sort<T>(values: &mut [T])
{
    for i in 1..values.len() {
        let mut key = values[i];
        let mut j = i - 1;
        while j > 0 && arr[j] > key {
            values.swap(j, j - 1);
            j = j - 1;
        }
        values[j + 1] = key
    }
}